### ISSUE TITLE
Regarding CY emission factors

### ISSUE THREAD

**Issue by silimotion** [electricitymaps/electricitymaps-contrib#6014]
## Description

Currently, Cyprus oil emission factor is 1125 gCO2eq/kWh. It seemed very high, so I checked the official emissions data from ETS and production from the Cyprus System Operator ([https://tsoc.org.cy/files/reports/annual-reports/TSOC_Annual_Report_2021.pdf](https://tsoc.org.cy/files/reports/annual-reports/TSOC_Annual_Report_2021.pdf)).
According to these data sources, the 3 oil power plants produced in 2021 a grand total of 3,077,688 kg CO2 and 4,337,147 kWh.

Which would mean a CO2 intensity of 709,6 g CO2/kWh.

I would edit the .yaml myself, but it seems some other values depend on this one and I want to be sure everything is done correctly.

Also note that the Balearic and Canary islands emissions could be calculated similarly (or with Electricity Maps own aggregated data).

So, is my method correct? If so, what needs to be changed exactly in the .yaml?

**Comment by VIKTORVAV99** [comment]
This calculations is pretty simple so feel free to do so.

Just make sure this value is only applied from 2021 and forward (earlier data can be added for each year and will be used when we recalculate things).

And if you have or can find official data for the Canary Islands, Balearic Islands (and maybe even the two Spanish autonomous cities in Africa?) feel free to do so for them as well. Preferably this would be a separate PR from the CY ones though, just to keep things simple.

**Comment by AJDelusion** [comment]
Are the emission totals from TSOC full life cycle emissions from oil (i.e. including oil extraction and processing) or just the direct emissions from the power plants? To be methodologically consistent with other factors would it not need to be the former?

**Comment by VIKTORVAV99** [comment]
> Are the emission totals from TSOC full life cycle emissions from oil (i.e. including oil extraction and processing) or just the direct emissions from the power plants? To be methodologically consistent with other factors would it not need to be the former?

The config files have both as we provide values for both in the API, however the app only use full lifecycle emissions.

**Comment by silimotion** [comment]
> Are the emission totals from TSOC full life cycle emissions from oil (i.e. including oil extraction and processing) or just the direct emissions from the power plants? To be methodologically consistent with other factors would it not need to be the former?

The emission totals come from EU-ETS so it should be consistent with what is done for other EU countries.

@VIKTORVAV99 I have read the .ipynb but I only found the computation for the direct emissions. How is the full lifecycle factor calculated from there?

**Comment by VIKTORVAV99** [comment]
@pierresegonne any input here? I'm not seeing any splits between the direct and lifecycle emissions either.

**Comment by pierresegonne** [comment]
To explain why we ended up with the original emission factor, one can see in the notebook:

<img width="679" alt="Screenshot 2023-10-24 at 09 02 10" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/32778266/348b76d8-c268-435b-af0b-d2fe90adcda7">

-> Which means that we used the avg EU as a fallback.

If the data is actually available, then let's update with the more granular data. 

The EU-ETS allows the computation of direct emission factors, we then add upstream emissions from our default. I just realised that it was not documented in the wiki, will add this as a TODO on our side.

The value for oil is 244 gCO2e/kWh (as seen as the difference between lifecyle and direct in https://github.com/electricitymaps/electricitymaps-contrib/wiki/Emission-factors)
